### PR TITLE
Doctests for basic agent functionality

### PR DESCRIPTION
**Pull request by PragTob**
Read the docs and thought it'd be nice to have usage examples
right there so why not make it doc tests :)

It's a bit repetetive as you always need to start the agent and
I figured even for start/start_link it'd be nice to show that
the value now lives there but that might be discussion worthy.

I also started out always calling Agent.stop to stop the agent
(the other agent tests do that) but the GenServer tests don't
seem to shut it down every time so I thought it might be okay
here in favor of less clutter and to have more clarity :)

Happy about feedback!

### CODE REVIEW

File: lib/elixir/lib/agent.ex [hunk 1]
```diff
@@ -150,6 +150,13 @@ defmodule Agent do
 
   If the given function callback fails with `reason`, the function returns
   `{:error, reason}`.
+
+  ## Examples
+
+      iex> {:ok, pid} = Agent.start_link fn -> 42 end
```
Comment (lexmag) [inline lib/elixir/lib/agent.ex]: Let's have parenthesis here and in all other places, we usually omit them only in DSL and macro calls.

File: lib/elixir/test/elixir/agent_test.exs [hunk 2]
```diff
@@ -2,6 +2,7 @@ Code.require_file "test_helper.exs", __DIR__
 
 defmodule AgentTest do
   use ExUnit.Case, async: true
+  doctest Agent
```
Comment (lexmag) [inline lib/elixir/test/elixir/agent_test.exs]: I think we commonly have newline between `use` and `doctest`.

**Comment by PragTob** [comment]
Updated :) Thanks for the review!

**Comment by lexmag** [comment]
Thank you @PragTob. 💛

**Pull request merged**